### ISSUE TITLE
[release/2.10] models got fail_accuracy on BMG WSL2

### ISSUE THREAD

**Issue by libohao1201** [intel/torch-xpu-ops#2656]
### 🐛 Describe the bug

The following timm models got fail_accuracy:

- sebotnet33ts_256: amp_bf16 training

- xcit_large_24_p8_224 : amp_fp16/fp16 training

- mixer_b16_224: fp16 training

- fbnetv3_b: bf16 training

```
Testing model sebotnet33ts_256
Testing with training mode.
loading model: 0it [00:07, ?it/s]
xpu  train sebotnet33ts_256
E1226 16:26:07.019000 564 torch/_dynamo/utils.py:3226] RMSE (res-fp64): 2.48998, (ref-fp64): 0.76574 and shape=torch.Size([64]). res.dtype: torch.float32, multiplier: 3.000000, tol: 0.010000, use_larger_multiplier_for_smaller_tensor: 0
E1226 16:26:07.019000 564 torch/_dynamo/utils.py:3087] Accuracy failed for key name stages.0.0.conv1_1x1.bn.bias.grad
fail_accuracy
Testing model xcit_large_24_p8_224
Testing with training mode.
loading model: 0it [00:10, ?it/s]
xpu  train xcit_large_24_p8_224
WARNING:common:fp64 golden ref were not generated for xcit_large_24_p8_224. Setting accuracy check to cosine
W1226 16:30:54.402000 758 torch/_dynamo/utils.py:3136] Similarity score=0.9684716463088989
E1226 16:30:54.404000 758 torch/_dynamo/utils.py:3087] Accuracy failed for key name patch_embed.proj.0.0.weight.grad
fail_accuracy
Testing model fbnetv3_b
Testing with training mode.
loading model: 0it [00:09, ?it/s]
xpu  train fbnetv3_b
E1226 16:32:32.298000 990 torch/_dynamo/utils.py:3226] RMSE (res-fp64): 0.01109, (ref-fp64): 0.00282 and shape=torch.Size([736]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.010000, use_larger_multiplier_for_smaller_tensor: 0
E1226 16:32:32.299000 990 torch/_dynamo/utils.py:3087] Accuracy failed for key name blocks.5.4.bn2.running_var
fail_accuracy
Testing model mixer_b16_224
Testing with training mode.
loading model: 0it [00:04, ?it/s]
xpu  train mixer_b16_224
E1226 16:32:56.622000 1133 torch/_dynamo/utils.py:3226] RMSE (res-fp64): 0.00431, (ref-fp64): 0.00109 and shape=torch.Size([768]). res.dtype: torch.float16, multiplier: 3.000000, tol: 0.010000, use_larger_multiplier_for_smaller_tensor: 0
E1226 16:32:56.623000 1133 torch/_dynamo/utils.py:3087] Accuracy failed for key name blocks.2.norm1.bias.grad
fail_accuracy
Testing model xcit_large_24_p8_224
Testing with training mode.
loading model: 0it [00:09, ?it/s]
xpu  train xcit_large_24_p8_224
WARNING:common:fp64 golden ref were not generated for xcit_large_24_p8_224. Setting accuracy check to cosine
W1226 16:36:06.610000 1258 torch/_dynamo/utils.py:3136] Similarity score=0.9846256375312805
E1226 16:36:06.612000 1258 torch/_dynamo/utils.py:3087] Accuracy failed for key name blocks.16.local_mp.conv2.weight.grad
fail_accuracy

```

Reproducer
```
bash inductor_xpu_test.sh timm_models amp_bf16 training accuracy xpu 0 static 1 0 sebotnet33ts_256
bash inductor_xpu_test.sh timm_models amp_fp16 training accuracy xpu 0 static 1 0 xcit_large_24_p8_224
bash inductor_xpu_test.sh timm_models bfloat16 training accuracy xpu 0 static 1 0 fbnetv3_b
bash inductor_xpu_test.sh timm_models float16 training accuracy xpu 0 static 1 0 mixer_b16_224
bash inductor_xpu_test.sh timm_models float16 training accuracy xpu 0 static 1 0 xcit_large_24_p8_224
```

### Versions

PyTorch version: 2.10.0+xpu (rc2 - 0084d20edc4cb40875a132db4caa8a720365bf3c)

**Comment by libohao1201** [comment]
Run with `TORCHINDUCTOR_ONLINE_SOFTMAX=0` still fail accuracy.

**Comment by chuanqi129** [comment]
@libohao1201 please also add driver version here. @kaileiyx please check whether the BMG native linux has such accuracy failure.

**Comment by libohao1201** [comment]
**xcit_large_24_p8_224**  is not supported in 2.11. For other cases, they still failed with "fail_accuracy" in pytorch 2.11 (BMG WSL - Ubuntu24.04).

**Comment by bjarzemb** [comment]
PT2.13 results:

- sebotnet33ts_256 amp_bf16 training: fail_accuracy
- xcit_large_24_p8_224 amp_fp16/fp16 training: not tested
- mixer_b16_224 fp16 training: pass
- fbnetv3_b bf16 training: fail_accuracy
